### PR TITLE
Removes `FromMont` and `ToMont` from `field.Element` api

### PR DESCRIPTION
**Pull request by gbotrel**
See #276

### CODE REVIEW

File: ecc/bls12-381/hash_vectors_test.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 func TestG1IsogenyVectors(t *testing.T) {
 	t.Parallel()
 
+	// TODO @gbotrel fix me test vectors shouldn't set words directly
```
Comment (Tabaie) [inline ecc/bls12-381/hash_vectors_test.go:12]: Right, if these come from the standard docs they should be available in decimal or hex form.

**Review by Tabaie — APPROVED** [review]

**Comment by samlaf** [comment]
Trying to upgrade our version of gnark-crypto from 0.8 to 0.9.
We were using
```
config := ecc.MultiExpConfig{ScalarsMont: true}
```
whereas now MultiExpConfig does not have a ScalarsMont argument.

I see that the default was set to false in 0.8. Can I just remove that field and everything will work fine? Or will this lead to breaking changes?

Disclaimer: I don't know what montgomery form is.

**Comment by gbotrel** [comment]
That was actually an overlook on my sight, pointed by other users in the past;

there is actually a subtle breaking change here.. 

the default is now effectively `ecc.MultiExpConfig{ScalarsMont: true}` so if you used that everywhere, you should be fine.

But callers without an explicit value would have had the default at: `ecc.MultiExpConfig{ScalarsMont: false}` and  the upgrade would be a silent change.. --> TLDR; all the APIs are now consistant and don't expose the underlying form / encoding of the field elements.

**Comment by samlaf** [comment]
> That was actually an overlook on my sight, pointed by other users in the past;
> 
> there is actually a subtle breaking change here..
> 
> the default is now effectively `ecc.MultiExpConfig{ScalarsMont: true}` so if you used that everywhere, you should be fine.
> 
> But callers without an explicit value would have had the default at: `ecc.MultiExpConfig{ScalarsMont: false}` and the upgrade would be a silent change.. --> TLDR; all the APIs are now consistant and don't expose the underlying form / encoding of the field elements.

Awesome thanks! Small note: would have been useful if you guys added a `Breaking Changes` section to the release notes.
Otherwise is there a good reference that you would recommend to understand these montgomery forms and how they are used in this library?

**Pull request merged**